### DISCUSSION TITLE
Tailwind Unable to resolve dynamic variable

### DISCUSSION THREAD

**Discussion by ruhul-appnap** [tailwindlabs/tailwindcss#11314] [Help]
**What version of Tailwind CSS are you using?**

v3.2.2

**What build tool (or framework if it abstracts the build tool) are you using?**

Next.js 12.2

**What version of Node.js are you using?**

v16

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows

**Reproduction URL**

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

A reproduction is **required** when filing an issue — any issue opened without a reproduction will be closed and you'll be asked to create a new issue that includes a reproduction. We're a small team and we can't keep up with the volume of issues we receive if we need to reproduce each issue from scratch ourselves.

**Describe your issue**
tailwind is unable to resolve variable dynamically. It happens in dev & production build both. Steps to produce
1. create a variable in extend
```
extend: {
      colors: {
        orderDelivered: "#2781EA",
     }
}
```
2. use this variable dynamically in code for the first time.
```
                <p className={`text-${orderStatusColor[order.order_status]}`}>
```

in browser we get 
```
<p class="text-orderDelivered">Delivered</p>
```

But this class is not loaded in browser. if we use this variable statically once then refresh the page we get the class and text gets its color.

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Reply by ruhul-appnap** [reply]
When i need to handle more cases like   

- placed,
- processing, 
- shipped, delivered,  
- cancelled, 
- returned 
 Code locks messy  and lost readability
